lib/Runtime/Library/TypedArray.cpp

```diff
@@ -509,12 +510,15 @@ namespace Js
             if (args.Info.Count > 3 && !JavascriptOperators::IsUndefinedObject(args[3]))
             {
                 mappedLength = ArrayBuffer::ToIndex(args[3], JSERR_InvalidTypedArrayLength, scriptContext, ArrayBuffer::MaxArrayBufferLength / elementSize, false);
+                newByteLength = mappedLength * elementSize;
 
-                if ((uint32)mappedLength > (byteLength - offset)/ elementSize)
+                if (offset + newByteLength > byteLength)
```

Comment:
I noticed a slight change in the spec while reviewing the indexing related changes. Please refer to the spec at https://tc39.github.io/ecma262/#sec-typedarray-buffer-byteoffset-length section 22.2.4.5 step 12.